pkg/controller/container-runtime-config/helpers.go

```diff
@@ -46,24 +50,43 @@ type tomlConfigStorage struct {
 	} `toml:"storage"`
 }
 
-// tomlConfig is another way of looking at a Config, which is
+// tomlConfigCRIOLogLevel is used for conversions when log-level is changed
 // TOML-friendly (it has all of the explicit tables). It's just used for
 // conversions.
-type tomlConfigCRIO struct {
+type tomlConfigCRIOLogLevel struct {
 	Crio struct {
-		crioconfig.RootConfig
-		API     struct{ crioconfig.APIConfig }     `toml:"api"`
-		Runtime struct{ crioconfig.RuntimeConfig } `toml:"runtime"`
-		Image   struct{ crioconfig.ImageConfig }   `toml:"image"`
-		Network struct{ crioconfig.NetworkConfig } `toml:"network"`
-		Metrics struct{ crioconfig.MetricsConfig } `toml:"metrics"`
+		Runtime struct {
+			LogLevel string `toml:"log_level,omitempty"`
+		} `toml:"runtime"`
 	} `toml:"crio"`
 }
 
-// ignitionConfig is a struct that holds the filepath and date of the various configs
+// tomlConfigCRIOPidsLimit is used for conversions when pids-limit is changed
+// TOML-friendly (it has all of the explicit tables). It's just used for
+// conversions.
+type tomlConfigCRIOPidsLimit struct {
```

Comment:
(This is OK, just a sigh:

I can’t say that I’m enthusiastic about maintaining a separate copy of the definitions; the dependency does not go away, it’s just harder to find.

OTOH the -475k lines is an extremely compelling argument, and there’s that comment about handling unset fields.)

Reply:
This definitely reduces the complexity with the dependency and all. Since it is only 3 fields, I am inclined to leave it as is for now.